src/layout/SigningStatusPanel/PanelSigning.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useRef } from 'react';
+import type { PropsWithChildren, ReactElement } from 'react';
+
+import { ErrorMessage, Heading, Modal, Paragraph } from '@digdir/designsystemet-react';
+
+import { Button } from 'src/app-components/Button/Button';
+import { Panel } from 'src/app-components/Panel/Panel';
+import { useIsAuthorised } from 'src/features/instance/ProcessContext';
+import { useProcessNavigation } from 'src/features/instance/ProcessNavigationContext';
+import { Lang } from 'src/features/language/Lang';
+import classes from 'src/layout/SigningStatusPanel/SigningStatusPanel.module.css';
+import { useNodeItem } from 'src/utils/layout/useNodeItem';
+import type { PanelProps } from 'src/app-components/Panel/Panel';
+import type { BaseLayoutNode } from 'src/utils/layout/LayoutNode';
+
+type SigningPanelProps = {
+  node: BaseLayoutNode<'SigningStatusPanel'>;
+  heading: string;
+  description?: string;
+  variant?: PanelProps['variant'];
+  actionButton: ReactElement<typeof Button>;
+  errorMessage?: string;
+};
+
+export function SigningPanel({
+  node,
+  heading,
+  description,
+  variant = 'info',
+  actionButton,
+  errorMessage,
+  children,
+}: PropsWithChildren<SigningPanelProps>) {
+  const canReject = useIsAuthorised()('reject');
+
+  return (
+    <Panel
+      variant={variant}
+      isOnBottom
+      className={classes.signingPanel}
+    >
+      <div className={classes.contentContainer}>
+        <Heading
+          level={4}
+          size='xs'
+        >
+          {heading}
+        </Heading>
+        {description && <Paragraph>{description}</Paragraph>}
+
+        {children}
+        <div>
+          <div className={classes.buttonContainer}>
+            {canReject && <Reject node={node} />}
+            {actionButton}
+          </div>
+          {errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
+        </div>
+      </div>
+    </Panel>
+  );
+}
+
+type RejectTextProps = {
+  node: BaseLayoutNode<'SigningStatusPanel'>;
+};
+
+function Reject({ node }: RejectTextProps) {
+  const modalRef = useRef<HTMLDialogElement>(null);
+  const { next, busy } = useProcessNavigation() ?? {};
+  const { textResourceBindings } = useNodeItem(node);
```

Comment:
If you're only accessing one property, the selector mode is better:
```suggestion
  const textResourceBindings = useNodeItem(node, (i) => i.textResourceBindings);
```

This will only re-render when the actual text resources change (not when any other property changes), so it's similar to `SELECT justThatFieldINeeded FROM table` instead of `SELECT * FROM table`.

Also, I hope we one day get to run expressions _here_ instead of in the hierarchy generator, and at that point selecting just what you need will hopefully be more future-proof.